### ISSUE TITLE
Different style for icon

### ISSUE THREAD

**Issue by SuperBo** [Alexays/Waybar#1119]
I try to mimic this theme in way bar https://raw.githubusercontent.com/adi1090x/files/master/polybar-themes/previews/blocks/main.gif.

However, I don't know how to style Icon different versus the text.

Does waybar support this option?

**Comment by soyuz2012** [comment]
I'm not sure that it's good solution, but you can try to create 2 different modules (one for icon and one for text), e.g. ``network`` and ``network#text``.

**Comment by rYR79435** [comment]
You can wrap your icon in a `<span>` and set a `background` attribute on it, like this:
```
"format": "{temperatureC}°C <span background=\"#f00\">{icon}</span>"
```
which results in:
![20210620_11h31m48s_grim](https://user-images.githubusercontent.com/60985157/122669071-df471800-d1bb-11eb-8ffa-97775d794211.png)

It's a bit of a hack, since your now styling from the config file and not the `style.css`. But you can nonetheless achieve what you're trying to do.
Btw. something like `class="icon"` won't work, because `class` is not valid pango markup. See [valid attributes here](https://developer.gnome.org/pygtk/stable/pango-markup-language.html)

**Comment by SuperBo** [comment]
yeah, thank you all. But I think it's a little hacky. I think waybar should provide us a official way to style icon in style.css.

**Comment by frogamic** [comment]
I agree, I don't think hardcoding the styles for icons in configuration is a viable solution. It would be difficult to allow styling "icons" specifically when you are able to put a fontawesome glyph anywhere in the format string though. It really is a shame pango spans cannot have classes.

**Comment by ashish-kus** [comment]
```
"backlight": {
                 "device": "intel_backlight",
                 "format": "{icon}",
                 "tooltip-format": "盛{Percentage}%",
                 "format-icons": ["", "", "", "", "" ],
                 "on-scroll-up" : "light -A 1",
                 "on-scroll-down" : "light -U 1"
             },
```

This is my backlight module, now what should i do to increase the size of icon in format. Kindly help.

**Comment by LukashonakV** [comment]
Hi @ashish-kus ,
Try this
```json
"backlight": {
                 "device": "intel_backlight",
                 "format": "<span size='9pt'>{icon}</span>",
                 "tooltip-format": "盛{Percentage}%",
                 "format-icons": ["", "", "", "", "" ],
                 "on-scroll-up" : "light -A 1",
                 "on-scroll-down" : "light -U 1"
             },
```

**Comment by Alexays** [comment]
Waybar can't target "the icon" from `style.css`, because to GTK a format string is just text + glyphs — there's no separate icon widget to select, and Pango spans can't carry CSS classes. The supported way is inline Pango markup in the `format`, styling the icon directly:
```jsonc
"backlight": { "format": "<span size='large' color='#8ab4f8'>{icon}</span> {percent}%" }
```
Or, as suggested above, split it into two modules (e.g. `network` + `network#text`) if you want full `style.css` control over each part. Valid Pango attributes: https://docs.gtk.org/Pango/pango_markup.html . Closing since that's the current answer — reopen if you'd like. 🙏
